go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,1130 @@
+		for _, pod := range pods {
+			// Ignore completed pods, host networked pods, pods not scheduled
+			if util.PodWantsHostNetwork(pod) || util.PodCompleted(pod) || !util.PodScheduled(pod) {
+				continue
+			}
+			ips, err := util.DefaultNetworkPodIPs(pod)
+			if err != nil {
+				return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods, fmt.Errorf("failed to get pod ips: %w", err)
+			}
+			if len(ips) == 0 {
+				continue
+			}
+			key := ktypes.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
+			selectedPods.Insert(key)
+			selectedPodIPs[key] = ips
+			podsNsName[key] = pod
+		}
+		selectedNamespacesPods[namespace.Name] = podsNsName
+		selectedNamespaces.Insert(namespace.Name)
+	}
+	if selectedPods.Len() == 0 {
+		return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods, nil
+	}
+	node, err := c.nodeLister.Get(c.nodeName)
+	if err != nil {
+		return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods,
+			fmt.Errorf("failed to find this node %q kubernetes Node object: %v", c.nodeName, err)
+	}
+	// max of 1 EIP IP is selected. Return when 1 is found.
+	for _, status := range eip.Status.Items {
+		if isValid := isEIPStatusItemValid(status, c.nodeName); !isValid {
+			continue
+		}
+		eIPNet, err := getFullMaskIPNet(status.EgressIP)
+		if err != nil {
+			return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods,
+				fmt.Errorf("failed to generate mask for EgressIP %s IP %s: %v", eip.Name, status.EgressIP, err)
+		}
+		isOVNManaged, err := util.IsOVNManagedNetwork(node, eIPNet.IP)
+		if err != nil {
+			return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods,
+				fmt.Errorf("failed to determine if EgressIP %s IP %s is OVN managed: %v", eip.Name, status.EgressIP, err)
+		}
+		if isOVNManaged {
+			continue
+		}
+		isV6 := eIPNet.IP.To4() == nil
+		found, link, err := findLinkToHostIP(eIPNet.IP, c.v4, c.v6)
+		if err != nil {
+			return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods,
+				fmt.Errorf("failed to find a network to host EgressIP %s IP %s: %v", eip.Name, status.EgressIP, err)
+		}
+		if !found {
+			return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods,
+				fmt.Errorf("expected to find network interface to assign egressIP %s IP %s", eip.Name, status.EgressIP)
+		}
+		// go through all selected pods and build a config per pod IP. We know there are at least one pod and these the
+		// pod(s) have IP(s).
+		eIPConfig = generateEIPConfigForPods(selectedPodIPs, link, eIPNet, isV6)
+		// ignore other EIP IPs. Multiple EIP IPs cannot be assigned to the same node
+		break
+	}
+	return eIPConfig, selectedNamespaces, selectedPods, selectedNamespacesPods, nil
+}
+
+func generateEIPConfigForPods(pods map[ktypes.NamespacedName][]net.IP, link netlink.Link, eIPNet *net.IPNet, v6 bool) eIP {
+	eIPConfig := newEIP()
+	eIPConfig.routeLink = getDefaultRouteForLink(link, v6)
```

Comment:
I think its better because routes are associated with links and to submit a route to route manger, you need to also include the link.

I want to refactor route manager interface and remove the custom Route struct that route manager provides to users and just use netlink.Route. Pat also said it above in a previous comment.